src/natpmp.cpp

```diff
@@ -170,18 +172,22 @@ bool natpmp::get_mapping(int index, int& local_port, int& external_port, int& pr
 }
 
 #ifndef TORRENT_DISABLE_LOGGING
+bool natpmp::should_log() const
+{
+	return m_callback.should_log_portmap(aux::portmap_transport::natpmp);
+}
+
 TORRENT_FORMAT(2, 3)
 void natpmp::log(char const* fmt, ...) const
 {
 	TORRENT_ASSERT(is_single_thread());
+	if (!should_log()) return;
```

Comment:
ideally, I should forward the variadic arguments until the very end in the alerts, but that requires more refactor